main.js

```diff
@@ -46,13 +44,25 @@ function displayNumbers(randomNumbers) {
                 }
             }
 
-            if (selectedOrder.length === sortedNumbers.length) {
+            if (allNumbersClicked(selectedOrder, sortedNumbers)) {
                 document.querySelector('#play_area').textContent = '';
             }
         });
     });
 }
 
+function numberHasBeenClicked(elementClass) {
+    return (!elementClass.contains('clicked'));
+}
+
+function isNextNumber(sortedNumbers, selectedOrder, numberBoxValue) {
+    return sortedNumbers[selectedOrder.length] === numberBoxValue;
```

Comment:
put brackets around the comparison to force it to happen before the return. some operations have lower precidence than return so it would return the first variable and then not do anything else.